lib/ruby_lsp/requests/document_symbol.rb

```diff
@@ -106,8 +106,36 @@ def visit_class(node)
         @stack.pop
       end
 
+      sig { override.params(node: SyntaxTree::MethodAddBlock).void }
+      def visit_method_add_block(node)
+        # TODO: extract commonality with parse_test
+
+        return unless node.call.message.value == "test" && node.call.arguments.is_a?(SyntaxTree::ArgParen)
+
+        full_name = node.call.arguments.arguments.parts.first.parts.map do |part|
```

Comment:
Are no elements in this entire chain nilable? It looks like a lot of potential for breaking.

Reply:
Agree, and I think `parse_test` is already doing this the right way.

Reply:
Yes I found there were some problems with this, I've added more tests cases and broke it down a bit. I'll aim to unify it with `parse_test`.